engine/classes/ElggTempFile.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * This class represents a physical file (by default in the system temp directory).
+ *
+ * Create a new \ElggTempFile object and optionaly specify a filename
+ *
+ * Open the file using the appropriate mode, and you will be able to
+ * read and write to the file.
+ *
+ * Trying to save this entity to the database will fail
+ *
+ * @see \ElggFile
+ */
+class ElggTempFile extends ElggFile {
+
+	/**
+	 * Set subtype to 'temp_file'.
+	 *
+	 * @return void
+	 */
+	protected function initializeAttributes() {
+		parent::initializeAttributes();
+
+		$this->attributes['subtype'] = 'temp_file';
+		$this->setFilename(uniqid());
+	}
+
+	/**
+	 * Return the system temp filestore based on the system temp directory.
+	 *
+	 * @return \ElggTempDiskFilestore
+	 */
+	protected function getFilestore() {
+		return _elgg_services()->temp_filestore;
+	}
+
+	/**
+	 * Transfer a file to a new owner and sets a new filename,
```

Comment:
why not use `@inheritdoc`?? save you on writing this all

Reply:
i needed to change the return value, which i can't do with `@inheritdoc`